lib/matplotlib/scale.py

```diff
@@ -265,6 +266,101 @@ def set_default_locators_and_formatters(self, axis):
             axis.set_minor_locator(NullLocator())
 
 
+class PowerTransform(Transform):
+    input_dims = output_dims = 1
+
+    def __init__(self, gamma, clip=False):
+        super().__init__()
+        self.gamma = gamma
+        self._clip = clip
+
+    def __str__(self):
+        return "{}(gamma={}, nonpositive={!r})".format(
+            type(self).__name__, self.gamma, "clip" if self._clip else "mask")
+
+    def transform_non_affine(self, a):
+        with np.errstate(divide="ignore", invalid="ignore"):
+            mask=np.ma.getmask(a)
+            d=np.asarray(a.data)
+            out = np.where(d>=0,np.power(d,self.gamma),d)
+            if self._clip:
+                out[d <= 0] = 0
+            mout=np.ma.masked_array(out,mask=mask)
```

Comment:
Formatting. There are many places where formatting is still wrong. Look into each changed line once. 
You could use the `Format Modified Lines` option in VSCode to automatically format lines only where you have made changes.

```suggestion
            mask = np.ma.getmask(a)
            d = np.asarray(a.data)
            out = np.where(d >= 0, np.power(d, self.gamma), d)
            if self._clip:
                out[d <= 0] = 0
            mout = np.ma.masked_array(out, mask=mask)
```